common/src/main/java/com/revenuecat/purchases/common/diagnostics/DiagnosticsManager.kt

```diff
@@ -69,14 +72,42 @@ class DiagnosticsManager(
 
     fun trackEvent(diagnosticsEvent: DiagnosticsEvent) {
         enqueue {
-            val anonymizedEvent = diagnosticsAnonymizer.anonymizeEventIfNeeded(diagnosticsEvent)
-            // WIP: Check that file size is not above certain limit. If it is, delete.
-            verboseLog("Tracking diagnostics event: $anonymizedEvent")
-            try {
-                diagnosticsFileHelper.appendEventToDiagnosticsFile(anonymizedEvent)
-            } catch (e: IOException) {
-                verboseLog("Error tracking diagnostics event: $e")
-            }
+            trackEventInCurrentThread(diagnosticsEvent)
+        }
+    }
+
+    private fun getEventsToSync(): List<JSONObject> {
+        val diagnosticsList = diagnosticsFileHelper.readDiagnosticsFile()
+        val diagnosticsInFileCount = diagnosticsList.size
+        if (diagnosticsInFileCount > MAX_NUMBER_EVENTS) {
+            val eventsToRemoveCount = diagnosticsInFileCount - MAX_NUMBER_EVENTS + 1
+            diagnosticsFileHelper.deleteOlderDiagnostics(eventsToRemoveCount)
+            trackMaxEventsStoredLimitReached(diagnosticsInFileCount, eventsToRemoveCount)
+            return diagnosticsFileHelper.readDiagnosticsFile()
+        }
+        return diagnosticsList
+    }
+
+    private fun trackMaxEventsStoredLimitReached(totalEventsStored: Int, eventsRemoved: Int) {
+        trackEventInCurrentThread(
+            DiagnosticsEvent.Log(
+                name = DiagnosticsLogEventName.MAX_EVENTS_STORED_LIMIT_REACHED,
+                properties = mapOf(
+                    "total_number_events_stored" to totalEventsStored,
+                    "events_removed" to eventsRemoved
+                )
+            )
+        )
+    }
+
+    private fun trackEventInCurrentThread(diagnosticsEvent: DiagnosticsEvent) {
```

Comment:
I separated this logic into this method to be able to send the `MAX_EVENTS_STORED_LIMIT_REACHED` event in the same sync we detect we've reached the limit. If we added tracking the event to the queue, we would sync the events that existed before, then add the event to the diagnostics file, so we would miss a sync for tracking these events.